File: Solutions/700-SearchBST/SearchBinarySearchTree.ts
import { TreeNode } from '../../types';

function searchBST(root: TreeNode | null, val: number): TreeNode | null {
    if (!root) return null;

    if (root.val === val) return root;

    return searchBST(root.left, val) || searchBST(root.right, val);
}

File: Solutions/700-SearchBST/SearchBinarySearchTree.java
import Types.TreeNode.*;

class Solution {
    public TreeNode searchBST(TreeNode root, int val) {
        if (root == null)
            return null;

        if (root.val == val)
            return root;

        TreeNode searchRight = searchBST(root.right, val);
        TreeNode searchLeft = searchBST(root.left, val);

        return searchRight == null ? searchLeft : searchRight;
    }
}